Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomValue.h>
+#include <AzCore/std/string/fixed_string.h>
+#include <AzCore/Name/Name.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    //! Defines a schema definition for a DocumentAdapter node.
+    //! Nodes have a name and metadata for validating their contents.
+    //! To register a Node you may inheirt from NodeDefinition and define traits:
+    //! struct MyNode : NodeDefinition
+    //! {
+    //!     static constexpr AZStd::string_view Name = "MyNode";
+    //!     static bool CanAddToParentNode(const Dom::Value& parentNode);
+    //! };
+    struct NodeDefinition
+    {
+        //! Defines the Name of the node definition.
+        //! This field must be defined for all NodeDefinitions.
+        static constexpr AZStd::string_view Name = "<undefined node name>";
+
+        //! If defined, specifies if this node may be added to a given parent (a DOM node).
+        //! By default all parent nodes are accepted.
+        static bool CanAddToParentNode([[maybe_unused]] const Dom::Value& parentNode)
+        {
+            return true;
+        }
+
+        //! If defined, specifies if this node may receive a child (a DOM Value that may be of an arbitrary type).
+        //! By default all values are accepted.
+        static bool CanBeParentToValue([[maybe_unused]] const Dom::Value& value)
+        {
+            return true;
+        }
+    };
+
+    //! Retrieves a node's name from a node definition.
+    //! \see NodeDefinition
+    template <typename TNodeDefinition>
```

Comment:
Nitpick: For most code names Hungarian Notation is not use, so it should be `NodeDefinition` without the `T`.

Reply:
I will forever and always write that extra `T` on autopilot when writing templates I think...